tests/unit_tests/test_validate_algorithm_request.py

```diff
@@ -0,0 +1,345 @@
+import json
+from unittest.mock import DEFAULT
+from unittest.mock import patch
+
+import pytest
+
+from mipengine.controller.api.exceptions import BadRequest
+from mipengine.controller.api.exceptions import BadUserInput
+from mipengine.controller.api.validator import validate_algorithm_request
+from tests.unit_tests.common_data_elements import (
+    common_data_elements as mocked_common_data_elements,
+)
+from tests.unit_tests.algorithms_specifications import (
+    algorithms_specifications as mocked_algorithms_specifications,
+)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def mock_cdes():
+    with patch(
```

Comment:
You should use `spec/autospec` in you mocks. Especially, `autospec=True` is very usefull.

Reply:
Added it for the node_catalog.

Cannot be used for CDEs/AlgoSpecs because I am providing the objects.